norse/torch/module/snn.py

```diff
@@ -0,0 +1,201 @@
+"""
+Base module for spiking neural network (SNN) modules.
+"""
+
+from typing import Callable, Optional, Tuple
+import torch
+
+from norse.torch.module.util import remove_autopses
+
+FeedforwardActivation = Callable[
+    # Input        State         Parameters       dt
+    [torch.Tensor, torch.Tensor, torch.nn.Module, float],
+    Tuple[torch.Tensor, torch.Tensor],
+]
+
+RecurrentActivation = Callable[
+    # Input        State         Input weights Rec. weights  Parameters       dt
+    [torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, torch.nn.Module, float],
+    Tuple[torch.Tensor, torch.Tensor],
+]
+
+
+class FeedforwardSNN(torch.nn.Module):
```

Comment:
Not sure about the name, but that's probably for later. It's more of a `Cell` as well, I think

Reply:
@cpehle also just had the idea that we could technically fuse the FF modules with the Cell modules if, for instance, allowed to set the weights to `None`. 

One option could also be to remove the non-temporal recurrent layer, so we could simply have an `SNNCell` and an `SNN` module. That would be much more in line with the PyTorch nomenclature. But I wonder if someone would be missing the non-temporal recurrent dynamics. For instance, in the case of running a sequence of modules for every timestep.